code/datums/status_effects/status_effect.dm

```diff
@@ -131,3 +131,119 @@
 		for(var/datum/status_effect/S in status_effects)
 			if(initial(S1.id) == S.id)
 				. += S
+
+//////////////////////
+// STACKING EFFECTS //
+//////////////////////
+
+/datum/status_effect/stacking
+	id = "stacking_base"
+	duration = -1 //removed under specific conditions
+	alert_type = null
+	var/stacks = 0 //how many stacks are accumulated, also is # of stacks that target will have when first applied
+	var/delay_before_decay //deciseconds until ticks start occuring, which removes stacks (first stack will be removed at this time plus tick_interval)
+	tick_interval = 10 //deciseconds between decays once decay starts
+	var/stack_decay = 1 //how many stacks are lost per tick (decay trigger) 
+	var/stack_threshold //special effects trigger when stacks reach this amount
+	var/max_stacks //stacks cannot exceed this amount
+	var/consumed_on_threshold = TRUE //if status should be removed once threshold is crossed
+	var/threshold_crossed = FALSE //set to true once the threshold is crossed, false once it falls back below
+	var/overlay_file
+	var/underlay_file
+	var/overlay_state // states in .dmi must be given a name followed by a number which corresponds to a number of stacks. put the state name without the number in these state vars
+	var/underlay_state // the number is concatonated onto the string based on the number of stacks to get the correct state name 
+	var/mutable_appearance/status_overlay
+	var/mutable_appearance/status_underlay
+
+/datum/status_effect/stacking/proc/threshold_cross_effect() //what happens when threshold is crossed
+	return
+
+/datum/status_effect/stacking/proc/stacks_consumed_effect() //runs if status is deleted due to threshold being crossed
+	return
+
+/datum/status_effect/stacking/proc/fadeout_effect() //runs if status is deleted due to being under one stack
+	return
+
+/datum/status_effect/stacking/proc/stack_decay_effect() //runs every time tick() causes stacks to decay
+	return
+
+/datum/status_effect/stacking/proc/on_threshold_cross()
+	threshold_cross_effect()
+	if(consumed_on_threshold)
+		stacks_consumed_effect()
+		qdel(src)
+
+/datum/status_effect/stacking/proc/on_threshold_drop()
+	return
+
+/datum/status_effect/stacking/proc/can_have_status()
+	if(owner.stat == DEAD)
+		return FALSE
+	else
+		return TRUE
+
+/datum/status_effect/stacking/proc/can_gain_stacks()
+	if(owner.stat == DEAD)
+		return FALSE
+	else
+		return TRUE
+
+/datum/status_effect/stacking/tick()
+	if(!can_have_status())
+		qdel(src)
+	else
+		add_stacks(-stack_decay)
+		stack_decay_effect()
+
+/datum/status_effect/stacking/proc/add_stacks(stacks_added)
+	if(stacks_added>0 && !can_gain_stacks())
+		return FALSE
+	owner.cut_overlay(status_overlay)
+	owner.underlays -= status_underlay
+	stacks += stacks_added
+	if(stacks>0)
```

Comment:
Would it kill ya to be consistent on whether you have space around your operators